src/api.js

```diff
@@ -88,8 +88,8 @@ function _scoord3d2Geometry(scoord3d, pyramid) {
     });
     return new PolygonGeometry([coordinates]);
   } else if (type === 'CIRCLE') {
-    let center = _scoord3dCoordinates2geometryCoordinates(scoord3d.centerCoordinates, pyramid);
-    let radius = scoord3d.radius;
+    let center = _scoord3dCoordinates2geometryCoordinates(data[0], pyramid);
+    let radius = data[1];
```

Comment:
I think this is incorrect. The second item of the array returned by `Circle.graphicData` is a point on the contour, not the radius. Both coordinates should actually be mapped from the slide coordinate space (Scoord3D) the map coordinate space (Geometry).

Initially, I thought it may be nicer to construct a circle with `centerCoordinates` and `radius`, rather than the coordinates of the center point and a point on the diameter. However, if we need to convert the coordinates back and forth repeatedly, it may be easier to just pass a `coordinates` array to the constructor similar to `Polyline`, `Polygon`, etc. What do you think @biharck?

Reply:
Hey @hackermd!
So, I changed the way we create the Circle to include the `flatCoordinates`:
https://github.com/dcmjs-org/dicom-microscopy-viewer/blob/event-strategy/src/scoord3d.js#L146

also changed the `_geometry2Scoord3d ` function to send the `flatCoordinates` chunked since `flatCoordinates` returns an array of size 4. I am chunking it into two arrays of size 3 (including 1 as z coord)
https://github.com/dcmjs-org/dicom-microscopy-viewer/blob/event-strategy/src/api.js#L74

finally, the `_scoord3d2Geometry` has changed as well to create the OpenLayers Circle.
https://github.com/dcmjs-org/dicom-microscopy-viewer/blob/event-strategy/src/api.js#L106

This way, OpenLayers handles the radius calculation for us.

What do you think @hackermd?

Reply:
Nice solution. I like it.

Reply:
We should fix the comments for `GraphicData`. Some of them are outdated and still reflect the 2D `Scoord`, e.g. https://github.com/dcmjs-org/dicom-microscopy-viewer/blob/9f3c9eecfb5d5e531ac56505a092bf3011789425/src/scoord3d.js#L150-L154

or https://github.com/dcmjs-org/dicom-microscopy-viewer/blob/9f3c9eecfb5d5e531ac56505a092bf3011789425/src/scoord3d.js#L185-L189

For `Circle`, we could also do an additional check in the constructor to ensure that the `coordinates` array has length 2.

Reply:
Thank you for the feedback @hackermd 
It is done. I tried to update the comments also include the test into the Circle constructor.